modules/gapi/test/common/gapi_video_tests_inl.hpp

```diff
@@ -131,7 +131,51 @@ TEST_P(BackgroundSubtractorTest, AccuracyTest)
     // Allowing 1% difference of all pixels between G-API and reference OpenCV results
     testBackgroundSubtractorStreaming(gapiBackSub, pOCVBackSub, 1, 1, learningRate, testNumFrames);
 }
+
+TEST_P(KalmanFilterTest, AccuracyTest)
+{
+    cv::gapi::video::KalmanParams kp(dDim, mDim, cDim, depth);
+
+    //measurement vector
+    cv::Mat measure_vec(mDim, 1, depth);
+    cv::randu(measure_vec, Scalar::all(-1), Scalar::all(1));
+
+    //control vector
+    cv::Mat ctrl_vec = Mat::zeros(cDim > 0 ? cDim : dDim, 1, depth);
+
+    if (cDim > 0)
+        cv::randu(ctrl_vec, Scalar::all(-1), Scalar::all(1));
+
+    // G-API Kalman state
+    cv::Mat gapiKState(dDim, 1, depth);
+    // OCV Kalman state
+    cv::Mat ocvKState(dDim, 1, depth);
+
+    // G-API code
+    cv::GMat m, ctrl;
+    cv::GOpaque<bool> have_m;
+    cv::GMat out = cv::gapi::KalmanFilter(m, have_m, ctrl, kp);
+    cv::GComputation comp(cv::GIn(m, have_m, ctrl), cv::GOut(out));
+
+    comp.apply(cv::gin(measure_vec, haveMeasure, ctrl_vec), cv::gout(gapiKState));
+
+    // OpenCV code
+    cv::KalmanFilter ocvKalman(dDim, mDim, cDim, depth);
+
+    ocvKState = cDim > 0 ? ocvKalman.predict(ctrl_vec) : ocvKalman.predict();
```

Comment:
I think we can just always pass `ctrl_vec` to `predict`, there is no harm if it's empty, Kalman can handle it

Reply:
Please see my answer for the same case above.

Reply:
Yes, I see now. Thanks!